text/0017-define_dataset_creation.md

```diff
@@ -0,0 +1,220 @@
+
+_It is possible to have multiple installations of Qri on your machine at once by manipulating the `$QRI_PATH` value. At time of writing (Oct. 2018), we haven't documented this properly._
+
+#### Git Commit Stores a snapshot, Qri Save Computes a snapshot
+while `qri save` and `git commit` are conceptually comparable, they're fairly different animals under the hood.
+
+Git is a _generic_ version control system. It's possible to version any binary data with git, with varying (incredible) degrees of success. The assumptions git makes about the information being versioned are very light (for example, git assumes many differences will occur across newline breaks).
+
+Git carries the assumption that the data provided to it is the _source of truth_. Git's job is to version things, the user's job is to tell git _what to version_.
+
+Like git, Qri's jobs is to version things. However, Qri is _not_ a generic version control system. Instead Qri only makes versions of a predefined dataset model. This association narrows the number of use cases in which Qri will function, but  delivers the benefit of making Qri _semantically versioned_. Because Qri is programmed to understand the intent & uses of various components of the dataset model, Qri can compute many important values on the user's behalf. This "acting on a user's behalf" comes from a long line of data managment practices. When managing data there is often too much information for the user to feasibly review each row
+
+Through the use of transform scripts, the user has tools to extend & control how a dataset is computed. Qri knows to run the transform in the first place because datasets have exactly one designated place for putting a transform script.
+
+#### Qri has no staging area
+While it is true that "qri save" is doing one less step than the git variant, git users will quickly point out that `git commit -am "changed stuff"` consolidates `git add` and `git commit` into a single step, so it's not much of a time-saver from a keystrokes perpective. Instead I've broken `git add` and `git commit` into two commands to show that Qri has no concept of a staging area.
+
+Instead of a staging area, Qri uses the concept of a _dry run_ to see what would happen without actually committing anything. 
+
+#### Qri registries are like git remotes
+With Qri it's possible to just run `qri save` and follow that with `qri conenct`, and now others will be able to see and consume your dataset.
+
+Qri Publish is a far less necessary step on 
+
+-- --
+
+## Save creates a dataset snapshot
+A dataset commit is a _snapshot_ of a dataset at a given point in time.
+
+Prepare - conform input data for saving
+Transform - execute a transform script with the
+Validate
+Save
+
+### Steps
+* **Prepare**
+  * screen for incorrect input
+  * check for required values
+  * remove unchanged input
+  * turn a _patch_ into an _update_
+* **Run Transform**
+  * transform scripts receive a _full snapshot_
```

Comment:
"full snapshot" of what? The changes (body, meta, etc) supplied by the user, right? We should specifically call out that the previous version's body is not the dataset's body, but can be retrieved by a helper function.